server/src/main/java/org/elasticsearch/action/ResolvedIndexExpressions.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.action;
+
+import java.util.Map;
+
+/**
+ * A collection of {@link ResolvedIndexExpression}, keyed by the original expression.
+ *
+ * <p>An example structure is:</p>
+ *
+ * <pre>{@code
+ * {
+ *   "my-index-*": {
+ *      {
+ *          "original": "my-index-*",
+ *          "localExpressions": {
+ *              "expressions": ["my-index-000001", "my-index-000002"],
+ *              "localIndexResolutionResult": "SUCCESS"
+ *          },
+ *          "remoteExpressions": ["remote1:my-index-*", "remote2:my-index-*"]
+ *      }
+ *   }
```

Comment:
Nit: Could we have more than one map entry for this example? This looks too much similar to the example of `ResolvedIndexExpression`.